File: src/main/scala/com/bombproj/framework/exception/NotFoundException.java
package com.bombproj.framework.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(HttpStatus.NOT_FOUND)
public class NotFoundException extends RuntimeException {

    public NotFoundException() {
        this("");
    }

    public NotFoundException(String message) {
        this(message, null);
    }

    public NotFoundException(String message, Throwable cause) {
        super(message, cause, false, false);
    }

    public NotFoundException(Throwable cause) {
        super("", cause, false, false);
    }
}

File: src/main/scala/com/bombproj/framework/exception/NotFoundException.scala
package com.bombproj.framework.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(HttpStatus.NOT_FOUND)
class NotFoundException (val message: String = "",
                         val cause: Throwable = null) extends RuntimeException(message, cause, false, false){

}
